vllm/spec_decode/spec_decode_worker.py

```diff
@@ -417,7 +417,10 @@ def execute_model(
         #    none of the requests in the batch have spec decoding enabled.
         # In any of these cases, the proposer and scorer workers
         # are called normally.
-        no_spec = num_lookahead_slots == 0 or disable_all_speculation or all(
```

Comment:
sure, so far I added tests for that `chunked_prefill=True` in `test_multistep_correctness.py` to make sure output matches because I thought those would be more reliable.

Reply:
Thanks for those tests. The reason why I was suggesting this unit test here is that it might be easier to test cases like the ones where the input contains non ending chunks and I am not sure if those cases are getting covered by the e2e tests.